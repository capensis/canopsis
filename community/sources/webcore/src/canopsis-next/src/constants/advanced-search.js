export const ADVANCED_SEARCH_ITEM_TYPES = {
  field: 'field',
  condition: 'condition',
  value: 'value',
  union: 'union',
};

export const ADVANCED_SEARCH_NEXT_ITEM_TYPES = {
  [ADVANCED_SEARCH_ITEM_TYPES.field]: ADVANCED_SEARCH_ITEM_TYPES.condition,
  [ADVANCED_SEARCH_ITEM_TYPES.condition]: ADVANCED_SEARCH_ITEM_TYPES.value,
  [ADVANCED_SEARCH_ITEM_TYPES.value]: ADVANCED_SEARCH_ITEM_TYPES.union,
  [ADVANCED_SEARCH_ITEM_TYPES.union]: ADVANCED_SEARCH_ITEM_TYPES.field,
};

export const ADVANCED_SEARCH_NOT = 'NOT';

export const ADVANCED_SEARCH_UNION_CONDITIONS = {
  and: 'AND',
  or: 'OR',
};

export const ADVANCED_SEARCH_CONDITIONS = {
  less: '<',
  more: '>',
  equal: '=',
  notEqual: '!=',
  like: 'LIKE',
  contains: 'CONTAINS',
};

export const ADVANCED_SEARCH_UNION_REGEXP_PATTERN = new RegExp(`\\s(${Object.values(ADVANCED_SEARCH_UNION_CONDITIONS).join('|')})(\\s|$)`, 'gi');

export const ADVANCED_SEARCH_UNION_FIELDS = [
  {
    value: ADVANCED_SEARCH_UNION_CONDITIONS.and,
    type: ADVANCED_SEARCH_ITEM_TYPES.union,
    text: ADVANCED_SEARCH_UNION_CONDITIONS.and,
  },
  {
    value: ADVANCED_SEARCH_UNION_CONDITIONS.or,
    type: ADVANCED_SEARCH_ITEM_TYPES.union,
    text: ADVANCED_SEARCH_UNION_CONDITIONS.or,
  },
];
