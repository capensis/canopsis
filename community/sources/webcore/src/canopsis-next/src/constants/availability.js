import { QUICK_RANGES } from './common';

export const AVAILABILITY_SHOW_TYPE = {
  percent: 0,
  duration: 1,
};

export const AVAILABILITY_QUICK_RANGES = {
  [QUICK_RANGES.last1Hour.value]: QUICK_RANGES.last1Hour,
  [QUICK_RANGES.last24Hour.value]: QUICK_RANGES.last24Hour,
  [QUICK_RANGES.today.value]: QUICK_RANGES.today,
  [QUICK_RANGES.yesterday.value]: QUICK_RANGES.yesterday,
  [QUICK_RANGES.last7Days.value]: QUICK_RANGES.last7Days,
  [QUICK_RANGES.previousWeek.value]: QUICK_RANGES.previousWeek,
  [QUICK_RANGES.previousMonth.value]: QUICK_RANGES.previousMonth,
  [QUICK_RANGES.thisMonth.value]: QUICK_RANGES.thisMonth,
  [QUICK_RANGES.thisMonthSoFar.value]: QUICK_RANGES.thisMonthSoFar,
  [QUICK_RANGES.last3Months.value]: QUICK_RANGES.last3Months,
  [QUICK_RANGES.last6Months.value]: QUICK_RANGES.last6Months,
};
