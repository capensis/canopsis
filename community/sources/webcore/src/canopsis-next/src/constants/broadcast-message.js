export const DEFAULT_BROADCAST_MESSAGE_COLOR = '#e75e40';

export const BROADCAST_MESSAGES_STATUSES = {
  active: 0,
  pending: 1,
  expired: 2,
};
