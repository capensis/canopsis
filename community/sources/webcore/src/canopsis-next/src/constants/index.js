export * from './recurrence-rule';
export * from './common';
export * from './entity';
export * from './alarm';
export * from './broadcast-message';
export * from './date';
export * from './dynamic-info';
export * from './event-filter';
export * from './pattern';
export * from './healthcheck';
export * from './idle-rule';
export * from './junit';
export * from './modal';
export * from './pbehavior';
export * from './permission';
export * from './remediation';
export * from './scenario';
export * from './snmp';
export * from './stats';
export * from './store';
export * from './widget';
export * from './kpi';
export * from './user';
export * from './service-weather';
export * from './map';
export * from './flowchart';
export * from './parameters';
export * from './declare-ticket';
export * from './link-rule';
export * from './external-data';
export * from './tag';
export * from './counter';
export * from './calendar';
export * from './icon';
export * from './availability';
