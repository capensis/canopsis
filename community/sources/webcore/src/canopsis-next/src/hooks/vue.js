import { getCurrentInstance } from 'vue';

/**
 * Hook for get current component instance
 *
 * @return {Vue}
 */
export const useComponentInstance = () => {
  const instance = getCurrentInstance();

  return instance?.proxy || instance;
};

/**
 * Hook for get current component options
 *
 * @return {Object}
 */
export const useComponentOptions = () => {
  const instance = useComponentInstance();

  return instance.$options;
};

/**
 * Hook for get current component model
 *
 * @return {Object}
 */
export const useComponentModel = () => {
  const { model } = useComponentOptions();

  return {
    event: model?.event ?? 'input',
    prop: model?.event ?? 'value',
  };
};
