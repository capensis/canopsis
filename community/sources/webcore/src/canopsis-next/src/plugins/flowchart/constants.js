export const DIRECTIONS = {
  northWest: 'nw',
  north: 'n',
  northEast: 'ne',
  east: 'e',
  southEast: 'se',
  south: 's',
  southWest: 'sw',
  west: 'w',
};
