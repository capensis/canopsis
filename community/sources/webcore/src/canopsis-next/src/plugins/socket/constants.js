export const REQUEST_MESSAGES_TYPES = {
  ping: 0,
  join: 1,
  leave: 2,
  authenticate: 3,
};

export const RESPONSE_MESSAGES_TYPES = {
  pong: 0,
  ok: 1,
  error: 2,
  close: 3,
  authenticated: 4,
};

export const ICONS_RESPONSE_MESSAGES_TYPES = {
  create: 0,
  update: 1,
  delete: 2,
};

export const MAX_RECONNECTS_COUNT = 10;

export const PING_INTERVAL = 5000;

export const RECONNECT_INTERVAL = 5000;

export const EVENTS_TYPES = {
  customClose: 'custom-close',
  closeRoom: 'close-room',
  networkError: 'network-error',
};

export const ERROR_MESSAGES = {
  authenticationFailed: 'authentication failed',
};
