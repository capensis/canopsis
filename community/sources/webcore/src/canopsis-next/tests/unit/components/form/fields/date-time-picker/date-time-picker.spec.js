import { generateShallowRenderer, generateRenderer } from '@unit/utils/vue';
import { mockDateNow } from '@unit/utils/mock-hooks';
import { DATETIME_FORMATS } from '@/constants';

import DateTimePicker from '@/components/forms/fields/date-time-picker/date-time-picker.vue';

const stubs = {
  'time-picker-field': true,
};

const factory = generateShallowRenderer(DateTimePicker, {

  stubs,
  listeners: {
    close: jest.fn(),
  },
});

const snapshotFactory = generateRenderer(DateTimePicker, {

  stubs,
  listeners: {
    close: jest.fn(),
  },
});

const selectButtons = wrapper => wrapper.findAll('v-btn-stub');
const selectCancelButton = wrapper => selectButtons(wrapper).at(0);
const selectApplyButton = wrapper => selectButtons(wrapper).at(1);
const selectTimePickerField = wrapper => wrapper.find('time-picker-field-stub');
const selectDatePicker = wrapper => wrapper.find('v-date-picker-stub');

describe('date-time-picker', () => {
  const nowTimestamp = 1386435600000;
  const nowDate = new Date(nowTimestamp);

  mockDateNow(nowTimestamp);

  test('Time updated after trigger time picker field', async () => {
    const wrapper = factory({
      propsData: {
        value: nowDate,
      },
    });

    const timePickerField = selectTimePickerField(wrapper);

    const newTime = '12:45';

    await timePickerField.vm.$emit('input', newTime);

    expect(timePickerField.attributes('value')).toBe(newTime);
  });

  test('Date updated after trigger date picker', async () => {
    const wrapper = factory({
      propsData: {
        value: nowDate,
      },
    });

    const datePicker = selectDatePicker(wrapper);

    const newDate = '2015-12-12';

    await datePicker.vm.$emit('input', newDate);

    expect(datePicker.attributes('value')).toBe(newDate);
  });

  test('Value changed after trigger apply button without changes', () => {
    const wrapper = factory({
      propsData: {
        value: nowTimestamp,
      },
    });

    const applyButton = selectApplyButton(wrapper);

    applyButton.vm.$emit('click');

    const inputEvents = wrapper.emitted('input');
    const closeEvents = wrapper.emitted('close');

    expect(closeEvents).toHaveLength(1);
    expect(inputEvents).toHaveLength(1);

    const [eventData] = inputEvents[0];
    expect(eventData.getTime()).toEqual(nowTimestamp);
  });

  test('Value changed after trigger apply button with changes', async () => {
    const wrapper = factory({
      propsData: {
        value: null,
      },
    });
    const timePickerField = selectTimePickerField(wrapper);
    const datePicker = selectDatePicker(wrapper);

    const newTime = '12:45';
    const newDate = '2015-12-12';
    const resultDate = new Date(`${newDate} ${newTime}`);

    await timePickerField.vm.$emit('input', newTime);
    await datePicker.vm.$emit('input', newDate);

    const applyButton = selectApplyButton(wrapper);

    applyButton.vm.$emit('click');

    const inputEvents = wrapper.emitted('input');

    expect(inputEvents).toHaveLength(1);

    const [eventData] = inputEvents[0];
    expect(eventData.getTime()).toEqual(resultDate.getTime());
  });

  test('Close emitted after trigger cancel button', () => {
    const close = jest.fn();
    const wrapper = factory({
      propsData: {
        value: null,
      },
      listeners: {
        close,
      },
    });

    const cancelButton = selectCancelButton(wrapper);

    cancelButton.vm.$emit('click');

    expect(close).toBeCalled();
  });

  test('Renders `date-time-picker` with default props', () => {
    const dateObject = new Date(nowTimestamp);
    const dateSpy = jest.spyOn(global, 'Date')
      .mockReturnValue(dateObject);

    const wrapper = snapshotFactory();

    expect(wrapper.element).toMatchSnapshot();

    dateSpy.mockClear();
  });

  test('Renders `date-time-picker` with custom props', async () => {
    const wrapper = snapshotFactory({
      propsData: {
        value: new Date(1387435600000),
        label: 'label',
        roundHours: true,
        dateFormat: DATETIME_FORMATS.long,
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });

  test('Renders `date-time-picker` without value', async () => {
    const wrapper = snapshotFactory({
      propsData: {
        value: null,
      },
      listeners: {
        close: jest.fn(),
      },
    });

    expect(wrapper.element).toMatchSnapshot();
  });
});
