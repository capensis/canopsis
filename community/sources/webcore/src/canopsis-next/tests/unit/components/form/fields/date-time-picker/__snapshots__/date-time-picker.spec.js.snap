// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`date-time-picker Renders \`date-time-picker\` with custom props 1`] = `
<div
  class="v-sheet theme--light"
>
  <div
    class="v-picker__title primary text-center"
  >
    <span
      class="text-h5"
    >
      label
    </span>
  </div>
  <div
    class="date-time-picker__body"
  >
    <div
      class="layout py-2 align-center justify-center"
    >
      <div
        class="flex v-date-time-picker__subtitle-wrapper"
      >
        <span
          class="v-date-time-picker__subtitle"
        >
          07/12/2013 18:00:00
        </span>
      </div>
      <div
        class="flex v-date-time-picker__subtitle-wrapper"
      >
        <time-picker-field-stub
          class="v-date-time-picker__subtitle"
          roundhours="true"
          stepsinhours="4"
          value="18:00"
        />
      </div>
    </div>
    <div>
      <div
        class="v-picker v-card v-picker--date theme--light"
      >
        <div
          class="v-picker__body v-picker__body--no-title theme--light"
          style="width: 290px;"
        >
          <transition-stub
            name="fade-transition"
          >
            <div>
              <div
                class="v-date-picker-header theme--light"
              >
                <button
                  aria-label="Previous month"
                  class="v-btn v-btn--icon v-btn--round theme--light v-size--default"
                  type="button"
                >
                  <span
                    class="v-btn__content"
                  >
                    <i
                      aria-hidden="true"
                      class="v-icon notranslate material-icons theme--light"
                    >
                      chevron_left
                    </i>
                  </span>
                </button>
                <div
                  class="v-date-picker-header__value"
                >
                  <transition-stub
                    name="tab-transition"
                  >
                    <div
                      class="primary--text"
                    >
                      <button
                        type="button"
                      >
                        December 2013
                      </button>
                    </div>
                  </transition-stub>
                </div>
                <button
                  aria-label="Next month"
                  class="v-btn v-btn--icon v-btn--round theme--light v-size--default"
                  type="button"
                >
                  <span
                    class="v-btn__content"
                  >
                    <i
                      aria-hidden="true"
                      class="v-icon notranslate material-icons theme--light"
                    >
                      chevron_right
                    </i>
                  </span>
                </button>
              </div>
              <div
                class="v-date-picker-table v-date-picker-table--date theme--light"
              >
                <transition-stub
                  name="tab-transition"
                >
                  <table>
                    <thead>
                      <tr>
                        <th>
                          S
                        </th>
                        <th>
                          S
                        </th>
                        <th>
                          S
                        </th>
                        <th>
                          S
                        </th>
                        <th>
                          S
                        </th>
                        <th>
                          S
                        </th>
                        <th>
                          S
                        </th>
                      </tr>
                    </thead>
                    <tbody>
                      <tr>
                        <td />
                        <td />
                        <td />
                        <td />
                        <td />
                        <td>
                          <button
                            class="v-btn v-btn--text v-btn--rounded theme--light"
                            type="button"
                          >
                            <div
                              class="v-btn__content"
                            >
                              7
                            </div>
                          </button>
                        </td>
                        <td>
                          <button
                            class="v-btn v-btn--text v-btn--rounded theme--light"
                            type="button"
                          >
                            <div
                              class="v-btn__content"
                            >
                              7
                            </div>
                          </button>
                        </td>
                      </tr>
                      <tr>
                        <td>
                          <button
                            class="v-btn v-btn--text v-btn--rounded theme--light"
                            type="button"
                          >
                            <div
                              class="v-btn__content"
                            >
                              7
                            </div>
                          </button>
                        </td>
                        <td>
                          <button
                            class="v-btn v-btn--text v-btn--rounded theme--light"
                            type="button"
                          >
                            <div
                              class="v-btn__content"
                            >
                              7
                            </div>
                          </button>
                        </td>
                        <td>
                          <button
                            class="v-btn v-btn--text v-btn--rounded theme--light"
                            type="button"
                          >
                            <div
                              class="v-btn__content"
                            >
                              7
                            </div>
                          </button>
                        </td>
                        <td>
                          <button
                            class="v-btn v-btn--text v-btn--rounded theme--light"
                            type="button"
                          >
                            <div
                              class="v-btn__content"
                            >
                              7
                            </div>
                          </button>
                        </td>
                        <td>
                          <button
                            class="v-btn v-date-picker-table__current v-btn--active v-btn--text v-btn--rounded theme--light primary"
                            type="button"
                          >
                            <div
                              class="v-btn__content"
                            >
                              7
                            </div>
                          </button>
                        </td>
                        <td />
                        <td />
                      </tr>
                    </tbody>
                  </table>
                </transition-stub>
              </div>
            </div>
          </transition-stub>
        </div>
      </div>
    </div>
  </div>
  <hr
    aria-orientation="horizontal"
    class="v-divider theme--light"
    role="separator"
  />
  <div
    class="layout mt-1 py-2 justify-space-around"
  >
    <button
      class="v-btn v-btn--text theme--light v-size--default"
      type="button"
    >
      <span
        class="v-btn__content"
      >
        Cancel
      </span>
    </button>
    <button
      class="primary v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default"
      type="button"
    >
      <span
        class="v-btn__content"
      >
        Apply
      </span>
    </button>
  </div>
</div>
`;

exports[`date-time-picker Renders \`date-time-picker\` with default props 1`] = `
<div
  class="v-sheet theme--light"
>
  <!---->
  <div
    class="date-time-picker__body"
  >
    <div
      class="layout py-2 align-center justify-center"
    >
      <div
        class="flex v-date-time-picker__subtitle-wrapper"
      >
        <span
          class="v-date-time-picker__subtitle"
        >
          07/12/2013
        </span>
      </div>
      <div
        class="flex v-date-time-picker__subtitle-wrapper"
      >
        <time-picker-field-stub
          class="v-date-time-picker__subtitle"
          stepsinhours="4"
          value="18:00"
        />
      </div>
    </div>
    <div>
      <div
        class="v-picker v-card v-picker--date theme--light"
      >
        <div
          class="v-picker__body v-picker__body--no-title theme--light"
          style="width: 290px;"
        >
          <transition-stub
            name="fade-transition"
          >
            <div>
              <div
                class="v-date-picker-header theme--light"
              >
                <button
                  aria-label="Previous month"
                  class="v-btn v-btn--icon v-btn--round theme--light v-size--default"
                  type="button"
                >
                  <span
                    class="v-btn__content"
                  >
                    <i
                      aria-hidden="true"
                      class="v-icon notranslate material-icons theme--light"
                    >
                      chevron_left
                    </i>
                  </span>
                </button>
                <div
                  class="v-date-picker-header__value"
                >
                  <transition-stub
                    name="tab-transition"
                  >
                    <div
                      class="primary--text"
                    >
                      <button
                        type="button"
                      >
                        December 2013
                      </button>
                    </div>
                  </transition-stub>
                </div>
                <button
                  aria-label="Next month"
                  class="v-btn v-btn--icon v-btn--round theme--light v-size--default"
                  type="button"
                >
                  <span
                    class="v-btn__content"
                  >
                    <i
                      aria-hidden="true"
                      class="v-icon notranslate material-icons theme--light"
                    >
                      chevron_right
                    </i>
                  </span>
                </button>
              </div>
              <div
                class="v-date-picker-table v-date-picker-table--date theme--light"
              >
                <transition-stub
                  name="tab-transition"
                >
                  <table>
                    <thead>
                      <tr>
                        <th>
                          S
                        </th>
                        <th>
                          S
                        </th>
                        <th>
                          S
                        </th>
                        <th>
                          S
                        </th>
                        <th>
                          S
                        </th>
                        <th>
                          S
                        </th>
                        <th>
                          S
                        </th>
                      </tr>
                    </thead>
                    <tbody>
                      <tr>
                        <td />
                        <td />
                        <td />
                        <td />
                        <td />
                        <td>
                          <button
                            class="v-btn v-btn--text v-btn--rounded theme--light"
                            type="button"
                          >
                            <div
                              class="v-btn__content"
                            >
                              7
                            </div>
                          </button>
                        </td>
                        <td>
                          <button
                            class="v-btn v-btn--text v-btn--rounded theme--light"
                            type="button"
                          >
                            <div
                              class="v-btn__content"
                            >
                              7
                            </div>
                          </button>
                        </td>
                      </tr>
                      <tr>
                        <td>
                          <button
                            class="v-btn v-btn--text v-btn--rounded theme--light"
                            type="button"
                          >
                            <div
                              class="v-btn__content"
                            >
                              7
                            </div>
                          </button>
                        </td>
                        <td>
                          <button
                            class="v-btn v-btn--text v-btn--rounded theme--light"
                            type="button"
                          >
                            <div
                              class="v-btn__content"
                            >
                              7
                            </div>
                          </button>
                        </td>
                        <td>
                          <button
                            class="v-btn v-btn--text v-btn--rounded theme--light"
                            type="button"
                          >
                            <div
                              class="v-btn__content"
                            >
                              7
                            </div>
                          </button>
                        </td>
                        <td>
                          <button
                            class="v-btn v-btn--text v-btn--rounded theme--light"
                            type="button"
                          >
                            <div
                              class="v-btn__content"
                            >
                              7
                            </div>
                          </button>
                        </td>
                        <td>
                          <button
                            class="v-btn v-date-picker-table__current v-btn--active v-btn--text v-btn--rounded theme--light primary"
                            type="button"
                          >
                            <div
                              class="v-btn__content"
                            >
                              7
                            </div>
                          </button>
                        </td>
                        <td />
                        <td />
                      </tr>
                    </tbody>
                  </table>
                </transition-stub>
              </div>
            </div>
          </transition-stub>
        </div>
      </div>
    </div>
  </div>
  <hr
    aria-orientation="horizontal"
    class="v-divider theme--light"
    role="separator"
  />
  <div
    class="layout mt-1 py-2 justify-space-around"
  >
    <button
      class="v-btn v-btn--text theme--light v-size--default"
      type="button"
    >
      <span
        class="v-btn__content"
      >
        Cancel
      </span>
    </button>
    <button
      class="primary v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default"
      type="button"
    >
      <span
        class="v-btn__content"
      >
        Apply
      </span>
    </button>
  </div>
</div>
`;

exports[`date-time-picker Renders \`date-time-picker\` without value 1`] = `
<div
  class="v-sheet theme--light"
>
  <!---->
  <div
    class="date-time-picker__body"
  >
    <div
      class="layout py-2 align-center justify-center"
    >
      <div
        class="flex v-date-time-picker__subtitle-wrapper"
      >
        <span
          class="v-date-time-picker__subtitle grey--text darken-1"
        >
          −−/−−/−−−−
        </span>
      </div>
      <div
        class="flex v-date-time-picker__subtitle-wrapper"
      >
        <time-picker-field-stub
          class="v-date-time-picker__subtitle"
          stepsinhours="4"
        />
      </div>
    </div>
    <div>
      <div
        class="v-picker v-card v-picker--date theme--light"
      >
        <div
          class="v-picker__body v-picker__body--no-title theme--light"
          style="width: 290px;"
        >
          <transition-stub
            name="fade-transition"
          >
            <div>
              <div
                class="v-date-picker-header theme--light"
              >
                <button
                  aria-label="Previous month"
                  class="v-btn v-btn--icon v-btn--round theme--light v-size--default"
                  type="button"
                >
                  <span
                    class="v-btn__content"
                  >
                    <i
                      aria-hidden="true"
                      class="v-icon notranslate material-icons theme--light"
                    >
                      chevron_left
                    </i>
                  </span>
                </button>
                <div
                  class="v-date-picker-header__value"
                >
                  <transition-stub
                    name="tab-transition"
                  >
                    <div
                      class="primary--text"
                    >
                      <button
                        type="button"
                      >
                        December 2013
                      </button>
                    </div>
                  </transition-stub>
                </div>
                <button
                  aria-label="Next month"
                  class="v-btn v-btn--icon v-btn--round theme--light v-size--default"
                  type="button"
                >
                  <span
                    class="v-btn__content"
                  >
                    <i
                      aria-hidden="true"
                      class="v-icon notranslate material-icons theme--light"
                    >
                      chevron_right
                    </i>
                  </span>
                </button>
              </div>
              <div
                class="v-date-picker-table v-date-picker-table--date theme--light"
              >
                <transition-stub
                  name="tab-transition"
                >
                  <table>
                    <thead>
                      <tr>
                        <th>
                          S
                        </th>
                        <th>
                          S
                        </th>
                        <th>
                          S
                        </th>
                        <th>
                          S
                        </th>
                        <th>
                          S
                        </th>
                        <th>
                          S
                        </th>
                        <th>
                          S
                        </th>
                      </tr>
                    </thead>
                    <tbody>
                      <tr>
                        <td />
                        <td />
                        <td />
                        <td />
                        <td />
                        <td>
                          <button
                            class="v-btn v-btn--text v-btn--rounded theme--light"
                            type="button"
                          >
                            <div
                              class="v-btn__content"
                            >
                              7
                            </div>
                          </button>
                        </td>
                        <td>
                          <button
                            class="v-btn v-btn--text v-btn--rounded theme--light"
                            type="button"
                          >
                            <div
                              class="v-btn__content"
                            >
                              7
                            </div>
                          </button>
                        </td>
                      </tr>
                      <tr>
                        <td>
                          <button
                            class="v-btn v-btn--text v-btn--rounded theme--light"
                            type="button"
                          >
                            <div
                              class="v-btn__content"
                            >
                              7
                            </div>
                          </button>
                        </td>
                        <td>
                          <button
                            class="v-btn v-btn--text v-btn--rounded theme--light"
                            type="button"
                          >
                            <div
                              class="v-btn__content"
                            >
                              7
                            </div>
                          </button>
                        </td>
                        <td>
                          <button
                            class="v-btn v-btn--text v-btn--rounded theme--light"
                            type="button"
                          >
                            <div
                              class="v-btn__content"
                            >
                              7
                            </div>
                          </button>
                        </td>
                        <td>
                          <button
                            class="v-btn v-btn--text v-btn--rounded theme--light"
                            type="button"
                          >
                            <div
                              class="v-btn__content"
                            >
                              7
                            </div>
                          </button>
                        </td>
                        <td>
                          <button
                            class="v-btn v-date-picker-table__current v-btn--rounded v-btn--outlined theme--light primary--text"
                            type="button"
                          >
                            <div
                              class="v-btn__content"
                            >
                              7
                            </div>
                          </button>
                        </td>
                        <td />
                        <td />
                      </tr>
                    </tbody>
                  </table>
                </transition-stub>
              </div>
            </div>
          </transition-stub>
        </div>
      </div>
    </div>
  </div>
  <hr
    aria-orientation="horizontal"
    class="v-divider theme--light"
    role="separator"
  />
  <div
    class="layout mt-1 py-2 justify-space-around"
  >
    <button
      class="v-btn v-btn--text theme--light v-size--default"
      type="button"
    >
      <span
        class="v-btn__content"
      >
        Cancel
      </span>
    </button>
    <button
      class="primary v-btn v-btn--is-elevated v-btn--has-bg theme--light v-size--default"
      type="button"
    >
      <span
        class="v-btn__content"
      >
        Apply
      </span>
    </button>
  </div>
</div>
`;
